pyquil/quil.py

```diff
@@ -188,7 +189,7 @@ def inst(self, *instructions):
 
         return self
 
-    def gate(self, name, params, qubits):
+    def gate(self, name: str, params: Iterable[ParameterDesignator], qubits: Iterable[Union[Qubit, QubitPlaceholder]]) -> 'Program':
```

Comment:
`Union[Qubit, QubitPlaceholder]` appears so often in quilbase.py that maybe it should get a name?